### PR TITLE
Generic doubly linked list.

### PR DESCRIPTION
**Pull request by AndreaOrru**
This implements a generic doubly linked list in the standard library.

### CODE REVIEW

File: std/linked_list.zig [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+const mem = @import("mem.zig");
+const Allocator = mem.Allocator;
+
+/// Generic doubly linked list.
+pub fn LinkedList(comptime T: type) -> type {
+    struct {
+        const List = this;
+
+        /// Node inside the linked list wrapping the actual data.
+        const Node = struct {
```
Comment (thejoshwolfe) [inline std/linked_list.zig]: shouldn't this be `pub`?

Comment (AndreaOrru) [inline std/linked_list.zig]: Probably. Using the list requires knowledge of the internals of `Node` anyway (the `data` field)

Comment (thejoshwolfe) [inline std/linked_list.zig]: I'm surprised this isn't a compile error.

Comment (AndreaOrru) [inline std/linked_list.zig]: @andrewrk you think this should be a compile error?

Comment (andrewrk) [inline std/linked_list.zig]: Where would the compile error be?

here?

```
    var one   = %%list.createNode(1);
```

error: `one` has type `Node` which is private
note: declared here

File: std/linked_list.zig [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+        }
+
+        /// Allocate and initialize a node and its data.
+        ///
+        /// Arguments:
+        ///     data: The data to put inside the node.
+        ///
+        /// Returns:
+        ///     A pointer to the new node.
+        pub fn createNode(list: &List, data: &const T) -> %&Node {
```
Comment (thejoshwolfe) [inline std/linked_list.zig]: error: cannot expose non-pub type Node in pub method signature.

File: std/linked_list.zig (hunk 1 above)
Comment (thejoshwolfe) [inline std/linked_list.zig]: the error should be on line 181

Comment (andrewrk) [inline std/linked_list.zig]: ok, I made #362 for this, I think we're still good to merge the PR

**Pull request merged**